File: fsharp/matrix/Matrix.fs
module Matrix

let rowsFrom (matrix:string) = matrix.Split('\n') |> Array.map (fun (r:string) -> r.Split(' ') |> Seq.map int |> Seq.toList)

let row index matrix = (rowsFrom matrix).[index]

let column index matrix = rowsFrom matrix |> Array.map (fun r -> r.[index]) |> Array.toList

File: python/matrix/matrix.py
class Matrix(object):

    def __init__(self, matrix_string):
        self.rows = []
        for line in matrix_string.split('\n'):
            row = [int(cell) for cell in line.split()]
            self.rows.append(row)

    def row(self, index):
        return self.rows[index-1]

    def column(self, index):
        return [row[index-1] for row in self.rows]